src/http/client.rs

```diff
@@ -248,7 +248,16 @@ impl EspHttpConnection {
 
         self.request_content_len = content_len.unwrap_or(0);
 
-        esp!(unsafe { esp_http_client_open(self.raw_client, self.request_content_len as _) })?;
+        esp!(unsafe {
+            esp_http_client_open(
+                self.raw_client,
+                if self.is_chunked_streaming_request(headers) {
```

Comment:
Remove the `if`. Assuming `is_chunked_streaming_request` is removed, and then unwrapping `content-len` just sets `request_content_len` to -1 if `content_len` is `None`, you'll no longer need it.